### PR TITLE
Add type definitions and refactor code

### PR DESCRIPTION
**Pull request by Americas**
Breaking change, bumps minimum node version from 6 to 8.

Breaking change, removes weird feature where a child logger could be enabled even if the debug string did not whitelist it, when a parent log is enabled and created first. This is not a consistent behavior, and could lead to unintended consequences. Furthermore, it is highly unlikely that this feature is being used purposefully.

Bumps package dependencies:
- bunyan@^1.8.15
- debug@^4.3.1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,13 +39,13 @@
...
-    "debug": "^2.2.0"
+    "bunyan": "^1.8.15",
+    "debug": "^4.3.1"
   },
   "devDependencies": {
     "uphold-scripts": "^0.4.0"
   },
   "engines": {
-    "node": ">=6"
+    "node": ">=8"
```
[the remark this answers is not in the document]
Comment (Americas) [reply package.json:49]: Done

[the remark this answers is not in the document]
Comment (Americas) [reply package.json:49]: @diogotorres97 Is travis CI even being used? :laughing:

[the remark this answers is not in the document]
Comment (diogotorres97) [reply package.json:49]: In public repositories, I think so

**Review by diogotorres97 — APPROVED** [review]
LGTM.

**Pull request merged**